Add docs `Hint` widget. (#144)
yiisoft/form: docs/hint.md

# Hint widget

Display hint for field form.

## Usage

```php
<?php

declare(strict_types=1);

namespace App\Form;

use Yiisoft\Form\FormModel;

final class TestForm extends FormModel
{
    private string $name = '';

    public function getAttributeHints(): array
    {
        return [
            'name' => 'Write your first name.',
        ];
    }    
}
```

Widget view:

```php
<?php

declare(strict_types=1);

use Yiisoft\Form\FormModelInterface;
use Yiisoft\Form\Widget\Field;
use Yiisoft\Form\Widget\Form;
use Yiisoft\Form\Widget\Hint;
use Yiisoft\Form\Widget\Text;

/**
 * @var FormModelInterface $data
 * @var object $csrf
 */
?>

<?= Form::widget()->action('widgets')->csrf($csrf)->begin() ?>

>     <?= Text::widget()->config($formModel, 'name') ?>
>     <?= Hint::widget()->config($formModel, 'name') ?>

Reviewer: Just a suggestion, but it looks interesting:

```suggestion
    <?= Text::widget()->for($formModel, 'name') ?>
    <?= Hint::widget()->for($formModel, 'name') ?>
```

Reads as "hint widget for $model's "name".

Reviewer: But that conflicts with https://github.com/yiisoft/form/pull/146/files#diff-b2dbc04239858c959223583f766058bc1685ade0b8ff4c9278cd3d30866bd5aaR119 :( Need a better name then...

    <hr class="mt-3">
    <?= Field::widget()->submitButton(['class' => 'button is-block is-info is-fullwidth', 'value' => 'Save']) ?>
<?= Form::end() ?>
```

That would generate the following code:

```html
<form action="widgets" method="POST" _csrf="BJKr0W43OfWETd0B8fZbrZFOKGzxkdoOZRRcuh4B1Gk3pdK_C2N6xfMbhXiQrjr153kZFrLLtXojcw6OV0CeAg==">
    <input type="hidden" name="_csrf" value="BJKr0W43OfWETd0B8fZbrZFOKGzxkdoOZRRcuh4B1Gk3pdK_C2N6xfMbhXiQrjr153kZFrLLtXojcw6OV0CeAg==">
    <input type="text" id="testform-name" name="TestForm[name]">
    <div>Write your first name.</div>
    <hr class="mt-3">
    <div>
        <input type="submit" id="submit-74174218703001" class="button is-block is-info is-fullwidth" name="submit-74174218703001" value="Save">
    </div>
</form>
```

### Custom hint text

> To configure a custom hint text, we simply change it in the widget, taking the previous example the code would be the following: 

Reviewer: ```suggestion
You can use custom hint text when calling the widget: 
```
